Feature: raw clipboard access (WIP)

livecodefraser opened the pull request.

Commit:
[[ JSON Library ]] Add release note

livecodefraser committed:
Merge pull request #2496 from runrevmark/feature-faster_sort

[[ Perf ]] Improve sort speed

runrevmark committed:
[[ ForeignCallbacks ]] Moved is_foreign bool to MCValueRef flags word to save space.

runrevmark committed:
[[ ForeignCallbacks ]] Updated LCB spec.

Commit:
[[ Release Notes ]] Make sure subsections are output in the order they are encountered

runrevmark committed:
[[ ForeignCallbacks ]] Added release note.

Commit:
[[ Bug 15848 ]] Fix overrelease bug in revXML

livecodefraser committed:
Merge pull request #2820 from livecodeali/feature-json_library

[[ Extensions ]] Add JSON library

livecodefraser committed:
Merge pull request #2821 from livecodeali/bugfix-release_notes_ordering

[[ Release Notes ]] Make sure subsections are output in the order they are encountered

runrevmark committed:
[[ ForeignCallbacks ]] Added native callback related tests.

livecodeali committed:
Merge pull request #2822 from livecodeali/bugfix-15848

[[ Bug 15848 ]] Fix overrelease bug in revXML

livecodefraser committed:
Merge pull request #2423 from runrevmark/lcb-native_callback

[[ ForeignCallbacks ]] Implement basic version of C callable LCB clos…

livecodefraser committed:
Merge pull request #2042 from peter-b/bugfix-14809

[Bug 14809] LCB: Remove deprecated type names.

Commit:
[[ Header Bar ]] Document header bar messages

Commit:
[[ Bug 15805 ]] Don't overwrite previously specified Syntax docs elements

livecodefraser committed:
Merge pull request #2823 from livecodeali/bugfix-15815

[[ Header Bar ]] Document header bar messages

Commit:
[[ Tree View ]] Fix loading of tree view properties

livecodeali committed:
Merge pull request #2824 from livecodeali/bugfix-15805

[[ Bug 15805 ]] Don't overwrite previously specified Syntax docs elements

livecodefraser committed:
Merge branch 'develop' of https://github.com/runrev/livecode into gyp-android_on_osx_fixes

Conflicts:
	config/paths.gypi

livecodefraser committed:
[[ Gyp ]] Remove references to src_top_dir_abs that crept back in

livecodefraser committed:
Merge pull request #2825 from livecodeali/feature-treeview_enhancements

[[ Tree View ]] Fix loading of tree view properties

Commit:
[[ Bug 15866 ]] Preserve internal empty lines when extracting docs elements

livecodefraser committed:
[[ Bugfix 15868 ]] Fix 'the mouseloc' on HTML5 to not always return 0,0

livecodefraser committed:
Merge pull request #2827 from livecodefraser/bugfix-15868

[[ Bugfix 15868 ]] Fix 'the mouseloc' on HTML5 to not always return 0,0

livecodefraser committed:
[[ Bugfix 15870 ]] HTML5: fix colour-swapping in embedded images

livecodefraser committed:
Merge pull request #2826 from livecodeali/bugfix-15866

[[ Bug 15866 ]] Preserve internal empty lines when extracting docs elements

Commit:
Merge commit '893540591f81037c2b617d10b0ee7c7e2a00fcff' into feature/scriptOnly_property

livecodefraser committed:
Merge pull request #2829 from livecodefraser/bugfix-15870

[[ Bugfix 15870 ]] HTML5: fix colour-swapping in embedded images

Commit:
[[ ScriptOnly ]] Added docs for scriptOnly property and documented create script only syntax.

livecodefraser committed:
Merge pull request #2710 from livecodefraser/gyp-android_on_osx_fixes

Fix various issues with building the Android engine on OSX

Commit:
[[ ScriptOnly ]] Added a note

livecodefraser committed:
Merge pull request #2800 from montegoulding/feature/scriptOnly_property

[[ ScriptOnly ]] Added RW scriptOnly property of stack

Commit:
[[ Tree View ]] Add icon to click for inspecting multiline or long elements

Commit:
[[ Tree View ]] Post message when double-clicked

livecodeian committed:
Merge branch 'develop' of https://github.com/runrev/livecode into bugfix-15129

livecodesebastien committed:
Merge pull request #2642 from livecodestephen/build-timestamps

Precisely preserve the timestamps when stripping debugging symbols

Commit:
[[ Bug 14970 ]] Correct tables in dictionary entry for HTMLText property

livecodeian committed:
[[ Bug 15345 ]] Fix call to MCContext::setorigin in sprite renderer callback to comply with changed behaviour.

LiveCodeSteven committed:
Update tan.lcdoc

As per bug report - http://quality.runrev.com/show_bug.cgi?id=14837 I have resolved the function error on line 40

livecodeian committed:
Merge branch 'develop' of https://github.com/runrev/livecode into canvas-effect_updates

LiveCodeSteven committed:
Create  bugfix-14837.md 

See reported bug here http://quality.runrev.com/show_bug.cgi?id=14837 
Replaced cosInDesgrees with tanInDegrees to match the opening name of the function.

LiveCodeSteven committed:
Rename  bugfix-14837.md to bugfix-14837.md

livecodeian committed:
[[ Canvas ]] Add export definition tag to MCCanvasEffectMake

livecodeian committed:
[[ Canvas ]] Add release note for effect constructor changes.

livecodeali committed:
Merge pull request #2850 from LiveCodeSteven/patch-1

[[ Bug 14837 ]] Update tan.lcdoc

livecodeali committed:
[[ Bug 15897 ]] Ensure LCB widget object is not nil before fetching save state

livecodeian committed:
[[ Bug 15129 ]] Add more info to bugfix note

livecodefraser committed:
Merge pull request #2223 from runrevian/canvas-effect_updates

Canvas effect updates

livecodeali committed:
Merge pull request #2848 from runrevian/bugfix-15345

[[ Bug 15345 ]] Fix call to MCContext::setorigin in sprite renderer c…

livecodeali committed:
Merge pull request #2852 from livecodeali/bugfix-15897

[[ Bug 15897 ]] Ensure LCB widget object is not nil before fetching save state

livecodeian committed:
[[ Bug 15129 ]] Quote code sections in bugfix release note

livecodeali committed:
Merge pull request #2847 from livecodeali/bugfix-14970

[[ Bug 14970 ]] Correct tables in dictionary entry for HTMLText property

livecodegeorgia committed:
[[Rewrite Navbar Widget]] Navbar widget now includes theme property that allows the user to select an Android or iOS style navbar

livecodefraser committed:
Add a mixin class providing basic reference-counting semantics

livecodefraser committed:
[[ RawClipboard ]] Initial in-engine API for raw clipboard access

livecodefraser committed:
Add a const-variant of MCAutoArray::operator[]

livecodefraser committed:
[[ RawClipboard ]] Initial implementation for OSX

Commit:
Update ide submodule reference

livecodefraser committed:
[[ RawClipboard ]] Add script hooks for the raw clipboard (getters only)

runrevmark commented on engine/src/exec-pasteboard.cpp:
```diff
@@ -290,6 +297,36 @@ void MCPasteboardEvalClipboardKeys(MCExecContext& ctxt, MCStringRef& r_string)
+    MCAutoRefcounted<const MCRawClipboardItem> t_item = t_clipboard->getItemAtIndex(0);
+    if (t_item == NULL)
+    {
+        r_string = kMCEmptyString;
```
You can do `r_string = MCValueRetain(kMCEmptyString)` here

livecodeali committed:
Merge pull request #2177 from runrevian/bugfix-15129

[[ Bug 15129 ]] implement "popup widget ..." as LCS syntax.

livecodestephen commented:
As `MCMixinRefcounted` is a generic template wouldn't it make sense to put it in a lower level library and not in engine?

runrevmark commented:
@livecodestephen: It could always be moved at a later date... The only 'lower-level' library we have at the moment is libfoundation and it doesn't quite sit there since the purpose of that library is MCValueRef types and related methods.

runrevmark commented:
This looks great - the only thing is a style one... In general, the C++ style I at least have always tried to adhere to is TitleCase for everything. So - GetItemAtIndex and MCAutoRefCounted.

livecodefraser commented:
I've been writing too much JavaScript-styled code recently ;)

livecodegeorgia committed:
[[Rewrite Navbar Widget]] Corrected backgroundOpacity docs, modified OnLoad() handler to account for existing saved navbars not having the theme property, modified setWidgetTheme handler to account for the fact that is an enum but can be set in script using any string, set initial theme to iOS and initial opacity to opaque

Commit:
[[ SVG Icon ]] Use svg icon editor for iconPathPreset property in the property inspector

livecodegeorgia committed:
[[Rewrite Navbar Widget]] Modified setOpacity handler to account for the fact that the opacity property is an enum that can be set in script, added release notes

livecodeali committed:
Merge pull request #2854 from runrev/community-docs

[[ Community Docs ]] Merge community-docs branch back into develop

Commit:
Merge remote-tracking branch 'upstream/develop-7.0' into develop

Conflicts:
	docs/dictionary/command/answer-effect.xml
	docs/dictionary/command/get.xml
	docs/dictionary/command/launch-url.xml
	docs/dictionary/command/mobileAddContact.xml
	docs/dictionary/command/mobileComposeHtmlMail.xml
	docs/dictionary/command/mobileComposeMail.xml
	docs/dictionary/command/mobileComposeUnicodeMail.xml
	docs/dictionary/command/mobileUpdateContact.xml
	docs/dictionary/command/open-socket.xml
	docs/dictionary/command/play-video.xml
	docs/dictionary/command/print-link.xml
	docs/dictionary/command/put-cookie.xml
	docs/dictionary/command/revBrowserAddJavaScriptHandler.xml
	docs/dictionary/command/revBrowserNavigate.xml
	docs/dictionary/command/revBrowserSet.xml
	docs/dictionary/command/revMail.xml
	docs/dictionary/command/revVideoGrabDialog.xml
	docs/dictionary/command/revXMLRPC_SetHost.xml
	docs/dictionary/command/secure-socket.xml
	docs/dictionary/function/QTEffects.xml
	docs/dictionary/function/QTVersion.xml
	docs/dictionary/function/mobileControlGet.xml
	docs/dictionary/function/revBrowserOpen.xml
	docs/dictionary/function/revBrowserOpenCef.xml
	docs/dictionary/object/AndroidBrowser.xml
	docs/dictionary/object/iosBrowser.xml
	docs/dictionary/property/dontUseQT.xml
	engine/Makefile.kernel-standalone
	engine/src/flst.h
	engine/src/sysosxrandom.cpp
	engine/src/sysunxrandom.cpp
	engine/src/sysw32random.cpp
	ide
	libcore/include/sserialize.h
	libcore/include/sserialize_lnx.h
	libcore/include/sserialize_osx.h
	libcore/include/sserialize_w32.h
	libcore/src/binary.cpp
	libcore/src/sserialize.cpp
	libcore/src/sserialize_lnx.cpp
	libcore/src/sserialize_osx.cpp
	libcore/src/sserialize_w32.cpp
	libgraphics/src/coretext.cpp
	libgraphics/src/legacyblendmodes.cpp
	libgraphics/src/legacygradients.cpp
	libgraphics/src/mblandroidtext.cpp
	libgraphics/src/path.cpp
	libgraphics/src/utils.cpp
	thirdparty

livecodegeorgia committed:
[[Rewrite Navbar Widget]] tweak to metadata and widgetTheme property doc

livecodegeorgia committed:
[[Rewrite Navbar Widget]] changed label metadata of property widgetTheme

Commit:
Update ide submodule pointer

livecodefraser committed:
[[ RawClipboard ]] Add clipboard locking and implement setting

livecodefraser committed:
[[ RawClipboard ]] Update OSX implementation to use locking and enable setting

livecodefraser committed:
[[ RawClipboard ]] Code style fixes

livecodeian committed:
[[ Bug 15908 ]] Fix android ffi closure crash. Update thirdparty submodule

livecodestephen committed:
Rename HACKING.md to CONTRIBUTING.md

CONTRIBUTING.md is clearer and hopefully less intimidating.  This also seems to be a more usual file name; it's used by rust, coffeescript, go, ruby, php, julia, scala, ....

livecodeali committed:
Merge pull request #2855 from livecodegeorgia/feature-rewrite_navbar_widget

[[Rewrite Navbar Widget]] Navbar widget now includes theme property t…

livecodeali committed:
Merge pull request #2867 from livecodestephen/contributing

Rename HACKING.md to CONTRIBUTING.md

livecodefraser committed:
[[ Emscripten ]] Show the LiveCode logo on the output HTML page

livecodefraser committed:
Merge pull request #2870 from livecodefraser/emscripten-livecode_logo

[[ Emscripten ]] Show the LiveCode logo on the output HTML page

peter-b committed:
[emscripten] Minor tweaks to standalone builder messages.

peter-b committed:
Merge branch 'develop' of github.com:runrev/livecode into js/lcb-extensions

livecodefraser committed:
Merge pull request #2863 from livecodeian/bugfix-15908

[[ Bug 15908 ]] Fix android ffi closure crash. Update thirdparty subm…

livecodeali committed:
Merge pull request #2859 from livecodeali/merge-develop-7.0_09.09.16

Merge develop 7.0 09.09.16

peter-b committed:
[emscripten] Avoid double slash (//) paths in zip files.

peter-b committed:
[emscripten] Avoid calling `fsync()` in order to prevent crash.

`fsync(2)` may pause when running under Emterpreter.  This causes a
crash because `MCEmscriptenFileHandle::Sync()` isn't compiled with
Emterpreter.

It's also probably unnecessary to call `fsync(2)` anyway; the
Emscripten filesystem is fully in-memory.

Commit:
Merge remote-tracking branch 'upstream/develop-7.0' into develop

Conflicts:
	docs/dictionary/function/time.xml
	engine/src/sysunxdate.cpp
	ide

Commit:
Update ide submodule reference

peter-b committed:
lc-compile: Make autogenerated functions return void.

This removes a large number of compilation warnings for Emscripten
builds with `-s LINKABLE=1`.

livecodeali committed:
Merge pull request #2871 from livecodeali/merge-develop-7.0_15.09.15

Merge develop 7.0 15.09.15

peter-b committed:
Merge pull request #2872 from peter-b/lcb/void-finalizers

lc-compile: Make autogenerated functions return void.

peter-b committed:
Merge pull request #2813 from peter-b/js/lcb-extensions

[emscripten] LCB extensions in HTML5 standalones

livecodestephen committed:
[[ CleanUp ]] Remove unneeded return statements from `OnSave` handlers

peter-b committed:
libfoundation: Add accessor functions for built-in type infos.

peter-b committed:
engine: Add accessor functions for built-in type infos.

peter-b committed:
[LCB] "foreign type" declarations bind to functions, not variables.

Change the meaning of the `foreign type <Name> binds to "<Symbol>"`
syntax.  `<Symbol>` is now expected to refer to a function with the
signature `MCTypeInfoRef (*)(void)`, which returns a type info
pointer.

This is required for platforms which can dynamically resolve symbols
referring to functions, such as Emscripten.

livecodefraser committed:
Add a non-retaining constructor for MCAutoValueRef classes

This constructor is only usable via explicit constructor syntax
(i.e `MCAutoValueRef t_value(p_value)` syntax).

peter-b committed:
Merge pull request #2874 from livecodestephen/unneeded-return

[[ CleanUp ]] Remove unneeded return statements from `OnSave` handlers

livecodefraser committed:
Remove some of the header insanity

object.h now includes the headers is requires rather than having
to add them in each place it is included.

livecodefraser committed:
Const-ify various MCAutoRefcounted accessors

livecodefraser committed:
Refactor a couple of methods in MCField taking MCStrings

livecodefraser committed:
[[ RawClipboard ]] Rename the MCClipboard syntax class

This has been renamed to MCClipboardProp to avoid clashes with
the new MCClipboard class used to provide the in-engine clipboard
interface.

livecodefraser committed:
[[ RawClipboard ]] Refactor the existing clipboard/drag code

All the clipboard and drag code now goes through MCRawClipboard.

peter-b committed:
Merge pull request #2873 from peter-b/js/no-fsync

[emscripten] Avoid calling `fsync()` in order to prevent crash.

livecodefraser committed:
[[ RawClipboard ]] OSX platform changes for the refactored clipboard code

Commit:
Update ide submodule reference

livecodefraser committed:
[[ RawClipboard ]] Remove the old clipboard code (generic and OSX for now)

livecodefraser committed:
[[ RawClipboard ]] OSX: only push changes if updates have been made

Commit:
[[ Release Notes ]] Update known issues

livecodefraser committed:
[[ RawClipboard ]] Keep the high-level and raw clipboards in sync

livecodefraser committed:
[[ RawClipboard ]] OSX: update the list of known type <-> UTI mappings

peter-b committed:
Merge pull request #2876 from peter-b/lcb/foreign-type-bindings

LCB: "foreign type" declarations bind to functions, not variables.

livecodeali committed:
Merge pull request #2877 from livecodeali/known_issues

[[ Release Notes ]] Update known issues

livecodefraser committed:
[[ RawClipboard ]] OSX: Add special-casing for "public.file-url" clipboard data

Commit:
[[ Bug 15941 ]] Fix circular script instance reference

livecodefraser committed:
[[ RawClipboard ]] Refactor MCClipboard::AddText slightly

livecodefraser committed:
[[ RawClipboard ]] Reject modifications to externally-provided clipboard data

livecodefraser committed:
[[ RawClipboard ]] Fix some minor typos

livecodefraser committed:
[[ RawClipboard ]] OSX: don't pull changes if clipboard is unchanged

livecodefraser committed:
[[ RawClipboard ]] OSX: AddRepresentation should replace existing reps

livecodefraser committed:
[[ RawClipboard ]] Implement getting and setting of file lists

livecodefraser committed:
Add a method to allow re-assigning MCAutoValueRefs

Commit:
Merge branch 'develop' of github.com:livecode/livecode into feature-treeview_resizable_view

livecodeali committed:
Merge pull request #2881 from livecodeali/bugfix-15941

[[ Bug 15941 ]] Fix circular script instance reference

Commit:
[[ Tree View ]] Add moveable separator to tree view

Commit:
Revert "Merge pull request #2642 from livecodestephen/build-timestamps"

This reverts commit c042606211b1142d87f50ecf27fab6a531618038, reversing
changes made to 21170c719d9c545cfdb5a988257ba56de94ff9ce.

livecodefraser committed:
[[ RawClipboard ]] Don't overwrite user 'text' when adding 'files'

Commit:
Merge remote-tracking branch 'upstream/develop-7.0' into merge-develop-7.0-17.09.15

Conflicts:
	Makefile
	ide-support/revdeploylibraryios.livecodescript
	libgraphics/libgraphics.gyp
	prebuilt/fetch-libraries.sh

livecodestephen committed:
The zip package is required to build on linux

Just doing a build on a fresh Debian Wheezy VM and I had to install the zip package.

livecodesebastien committed:
Merge pull request #2884 from livecodeali/merge-develop-7.0-17.09.15

Merge develop 7.0 17.09.15

Commit:
[[ Tree View ]] Add release note

peter-b committed:
Merge pull request #2885 from livecodestephen/install-zip

The zip package is required to build on linux

Commit:
[[ Segmented Control ]] Rejig the documentation of segmented control properties

Commit:
[[ Segmented Control ]] Use editorlist editor for icons when available

Commit:
[[ Segmented Control ]] Use editorlist editor for icons when available

Commit:
[[ Segmented Control ]] Add segmentCount property

Commit:
[[ Segmented Control ]] Use numeric editor for selected segment; make 0 default

livecodegeorgia committed:
[[Rewrite widget]]

livecodegeorgia committed:
[[Rewrite Widget]] no longer returning the properties array in the OnSave() handler

livecodegeorgia committed:
[[Rewrite Widget]] added borders to thumb where necessary in android theme

livecodegeorgia committed:
Merge branch 'develop' of https://github.com/runrev/livecode into feature-rewrite_switchbutton_widget

livecodegeorgia committed:
[[Rewrite Widget]] tightened up Android themes - drop shadows, borders and pressed state - corrected typos

livecodegeorgia committed:
[[Rewrite Widget]] modified set handlers for widgetTheme and colorScheme properties

livecodegeorgia committed:
[[Rewrite Widget]] iOS switch button is always enabled

livecodegeorgia committed:
[[Rewrite Widget]] improved appearance of iOS switch button - drop shadow, border, shape and size

livecodegeorgia committed:
[[Rewrite Widget]] added mouse move handler

Commit:
[[ Segmented Control ]] Rejig the way selections work in the segmented control; document messages it posts

Commit:
[[ Segmented Control ]] Make default names consistent

livecodegeorgia committed:
[[Rewrite widget]] updated set handlers, added a mouse release handler, updated mouse up handler, added documentation for switchChanged message

Commit:
[[ Docs ]] Don't output docs error when public handlers in widgets have no docs

Commit:
[[ Builder ]] Ensure docs dirs are created now that they are no longer in repo

livecodeali committed:
Merge pull request #2892 from livecodeali/docs-widget_public_handlers

[[ Docs ]] Don't output docs error when public handlers in widgets have no docs

livecodegeorgia committed:
[[Rewrite Widget]] updated clauses so that widget drawing only depends on the value of variable mSwitchIsInOnPosition

livecodeali committed:
Merge pull request #2893 from livecodeali/bugfix-docs_path

[[ Builder ]] Ensure docs dirs are created now that they are no longer in repo

livecodefraser committed:
[[ RawClipboard ]] Fix a Win32 compilation error

livecodefraser committed:
[[ RawClipboard ]] Change the API for encoding file lists

livecodefraser committed:
Add MCAutoDataRefArray

livecodefraser committed:
[[ RawClipboard ]] Fix a pointer cast in the refactored field code

livecodefraser committed:
[[ RawClipboard ]] Implementation for Win32

Commit:
[[ Cleanup ]] Remove unused binary builder stack

Commit:
Update ide submodule ptr

livecodefraser committed:
[[ RawClipboard ]] Fix mac server building

peter-b committed:
[emscripten] libscript: Must dlopen() even the default module.

`RTLD_DEFAULT` doesn't work as an argument to `dlsym()` on Emscripten.
The default module must always be opened explicitly.

peter-b committed:
[emscripten] Run startup script in temporary stack.

Currently, widgets are instantiated on stack load.  This means that
all extensions required by a stack must be loaded with `load
extension` *before* attempting to load the stack itself.

This means that the emscripten standalone `__startup` script needs to
be run on a temporary `MCStack` instance *before* attempting to load
the standalone's main stack.

peter-b committed:
[emscripten] build: tweak cc flags to allow dynamic binding.

This is needed to allow LiveCode Builder code to bind to core library
and engine functions via `dlsym(3)`.  There are several components:

* `-rdynamic` ensures that public symbols are dynamically linkable

* `-s LINKABLE=1` makes Emscripten output a symbol table in the JS

* `-s RESERVED_FUNCTION_POINTERS=1024` creates a lookup table for
  function pointers.

Because the number of flags passed to *emcc* is getting quite long,
I've made sure that in both places they occur they're in the same
order.

peter-b committed:
[emscripten] Implement ffi_call() and ffi_prep_cif_machdep().

This patch enables running the LiveCode Builder virtual machine
(libscript) in the browser.

It adds a very naive, Emscripten-specific implementation of
`ffi_prep_cif_machdep()` and `ffi_call()`, using inline JavaScript
code.  The implementations are embedded in libfoundation, in order to
avoid needing to add an additional link library.

* `ffi_prep_cif_machdep()` does nothing.

* `ffi_call()` computes the correct marshalling rules for each
  function from scratch for every invocation, which is probably slow.

peter-b committed:
[Tests] Add livecodescript test infrastructure.

Add a test suite infrastructure for `.livecodescript` files.

peter-b committed:
Merge pull request #2897 from peter-b/js/lcb-extensions

[emscripten] Enable running the LiveCode Builder virtual machine

peter-b committed:
[emscripten] Improved JavaScriptification script in Python.

Replace the existing `emscripten-javascriptify` with a new one written
in Python.  This is useful for commercial builds, because it makes it
easier to pass in arbitrary numbers of both preamble JS and library JS
files to the compiler.

I considered doing this in Bash but decided it would be faster and
less error-prone to just port it to Python.

Commit:
[[ Segmented Control ]] Use constants for default label and icon

Commit:
[[ Bug ]] Trim whitespace before counting number of lines

livecodeali committed:
Merge pull request #2895 from livecodeali/cleanup

[[ Cleanup ]] Remove unused binary builder stack

peter-b committed:
[emscripten] Split overlay logic into separate preamble script.

Now that the JavaScript-ification step can handle multiple preamble
files, we can put the overlay logic into a separate .js file.  Yay
factoring out unrelated code into separate files!

livecodegeorgia committed:
[[Rewrite Widget]] added theme property

livecodegeorgia committed:
[[Rewrite widget]] removed enabled property in the widget

livecodegeorgia committed:
Merge pull request #2890 from livecodegeorgia/feature-rewrite_switchbutton_widget

[[ Switch Button ]] Added Android themes

livecodeali committed:
Merge pull request #2858 from livecodeali/feature-preset_path_editor

[[ SVG Icon ]] Use svg icon editor for iconPathPreset property in the property inspector

Commit:
[[ Json Library ]] Add to installer manifest

livecodegeorgia committed:
[[Rewrite widget]] added android properties

livecodesebastien committed:
[[ RunRevReferences ]] Remove references to RunRev in LiveCode 8 files copyright

livecodeali committed:
Merge pull request #2901 from livecodeali/docs-release_notes

[[ Bug ]] Trim whitespace before counting number of lines in release note

livecodesebastien committed:
[[ RunRevReferences ]] Update .js and .mlc files

livecodesebastien committed:
[[ RunRevReferences ]] Remove unusually formatted copyright

livecodegeorgia committed:
[[Rewrite widget]] added variables and constants in order to be able to draw the android header

livecodesebastien committed:
[[ RunRevReferences ]] Remove references to 'Revolution' and 'Revolution custom library' from lcdoc files

livecodesebastien committed:
[[ RunRevReferences ]] Update remaining files in the docs folder

livecodesebastien committed:
[[ RunRevReferences ]] Update toolchain/lc-compile/src copyrights in .g files

peter-b committed:
libscript: Fix some sometimes-unitialised warnings in release mode.

peter-b committed:
libscript: Avoid warnings from empty macros in block scope.

peter-b committed:
libfoundation: Enable debug logging if DEBUG_LOG is defined.

This means that `MCLog()` can be enabled even if making a release
build by defining `DEBUG_LOG`.  `DEBUG_LOG` is implied by `_DEBUG`.

peter-b committed:
libfoundation: Make sure MCHandlerTypeLayout.abi is initialised.

peter-b committed:
tests: Allow LCB testsuite to be run under valgrind.

To run the tests under valgrind, use e.g.

    VALGRIND="valgrind --track-origins=yes" make -C tests/lcb

peter-b committed:
libscript: Clean up on-stack temporary storage for foreign invoke.

When getting foreign handler invocation working for the emscripten
port, there were some problems found with stack corruption involving
the temporary stack storage used by `MCScriptPerformForeignInvoke`.

The bug was due to Emscripten needing pointers to be 32-bit aligned.

This patch:

* Introduces a new `__MCScriptStackStorage` template class for the
  on-stack storage, with requests going through an `Allocate()`
  method.

* Ensures stack storage allocations are pointer-aligned.

* Adds assertions to catch storage over-allocation.

peter-b committed:
[emscripten] Add (disabled) debug logging to ffi_call().

This patch adds some (commented-out) debugging code to the embedded
JavaScript that implements `ffi_call()`.

livecodesebastien committed:
Revert "[[ RunRevReferences ]] Remove references to 'Revolution' and 'Revolution custom library' from lcdoc files"

This reverts commit 655c479309a86968139f0508f0c4b60eb6cf8091.

livecodesebastien committed:
[[ RunRevReferences ]] Properly update '<>' entries with Revolution / Revolution custom library

livecodesebastien committed:
[[ RunRevReferences ]] Update glossary references to Revolution / Revolution custom library in .lcdoc files

livecodesebastien committed:
[[ RunRevReferences ]] Replace <LiveCode|LiveCode> with <LiveCode> in lcdocs

livecodesebastien committed:
Merge pull request #2908 from livecodesebastien/remove_runrev_references

[[ RunRevReferences ]] Remove references to RunRev in LiveCode 8 file…

Commit:
[[ Cleanup ]] Remove no longer used 'modular' directory

livecodeali committed:
Merge pull request #2905 from livecodeali/bugfix-json_lib_install

[[ Json Library ]] Add to installer manifest

livecodegeorgia committed:
[[Rewrite Widget]] draw android header - background, left action and title

livecodefraser committed:
Fix the Linux Poll implementation to also poll GLib's FDs

Previously, these weren't getting included when polling so some
UI events were not being delivered in a timely manner.

livecodefraser committed:
[[ RawClipboard ]] Fix various refcounting bugs

livecodefraser committed:
Add a non-retaining setter to MCAutoValueRef

livecodefraser committed:
[[ RawClipboard ]] Implement for Linux

livecodeali committed:
Merge pull request #2911 from livecodeali/cleanup

[[ Cleanup ]] Remove no longer used 'modular' directory

livecodefraser committed:
[[ RawClipboard ]] Linux: add multiple formats for each type

This is required as various apps appear to look for different
strings for each type (for example, there are 3 different MIME
types used to represent RTF).

livecodefraser committed:
[[ RawClipboard ]] Linux: wait for response before getting clipboard data

The X11 clipboard model is asynchronous so we need to wait for
the clipboard owner to send us the data before attempting to read
it.

livecodefraser committed:
[[ RawClipboard ]] Linux: work around a bug when pre-allocating MCDataRef space

livecodefraser committed:
[[ RawClipboard ]] Linux: support some special clipboard types

livecodefraser committed:
[[ RawClipboard ]] Linux: make sure we return the TARGETS list with the correct type

peter-b committed:
Merge pull request #2913 from peter-b/tests/livecodescript-testsuite

[Tests] Add livecodescript test infrastructure.

Commit:
[[ Bug 16006 ]] Output a type field to a module manifest

Commit:
Add bugfix note

Commit:
Merge remote-tracking branch 'upstream/develop-7.0' into merge-develop_7.0_24.09.15

Conflicts:
	Installer/package.txt
	engine/src/lnxstack.cpp
	ide
	version

peter-b committed:
Merge pull request #2921 from peter-b/js/javascriptify

[emscripten] Improved JavaScript-ification script in Python.

Commit:
[[ Bug 16009 ]] Menu arrays come from script, so default values must be strings

livecodeali committed:
Merge pull request #2922 from livecodeali/merge-develop_7.0_24.09.15

Merge develop 7.0 24.09.15

livecodeali committed:
Merge pull request #2918 from livecodeali/bugfix-16006

[[ Bug 16006 ]] Output a type field to a module manifest

livecodefraser committed:
[[ RawClipboard ]] Linux: fix issues with the drag-and-drop loop

livecodefraser committed:
[[ RawClipboard ]] Update the middle-click clipboard before pasting

livecodefraser committed:
[[ RawClipboard ]] Linux: fix incompatibilities with external drag-and-drop sources

livecodegeorgia committed:
[[Rewrite Widget]] revisited code to draw iOS header

livecodegeorgia committed:
[[Rewrite Widget]] added android property distinctTitle

livecodegeorgia committed:
[[Rewrite Widget]] improved drawing of title and left icon in Android

peter-b committed:
Merge pull request #2928 from peter-b/js/overlay-preamble

[emscripten] Split overlay logic into separate preamble script.

livecodegeorgia committed:
[[Rewrite Widget]] added handler to draw android actions (WIP)

livecodegeorgia committed:
[[Rewrite Widget]] modified handler to draw android actions (WIP)

Commit:
[[ Bug 15981 ]] Remove OnMouseStillDown from widget docs

livecodeali committed:
Merge pull request #2931 from livecodeali/bugfix-15981

[[ Bug 15981 ]] Remove OnMouseStillDown from widget docs

livecodegeorgia committed:
[[Rewrite Widget]] corrected setting of dependent properties in Android, added handler to draw the iOS and Android actions

peter-b committed:
Merge pull request #2929 from peter-b/js/lcb-extensions

Ensure pointer alignment for temporary storage in libscript FFI calls.

livecodegeorgia committed:
[[Rewrite Widget]] added release notes, added additional details to property dics

livecodefraser committed:
[[ RawClipboard ]] Fix an off-by-1 error when pasting files to the OSX clipboard

livecodefraser committed:
[[ RawClipboard ]] Linux: robustness fixes when putting files on the clipboard

livecodegeorgia committed:
[[Rewrite widget]] modified hadnler that sets the color scheme of the android widget

livecodefraser committed:
[[ RawClipboard ]] Don't clear the selection clipboard when a field gains focus

livecodeali committed:
Merge pull request #2887 from livecodeali/feature-treeview_resizable_view

Feature treeview resizable view

livecodeali committed:
Merge pull request #2888 from livecodeali/feature-friendlier_segmented_control

Feature friendlier segmented control

livecodefraser committed:
[[ RawClipboard ]] OSX: fix handling of file paths

livecodegeorgia committed:
[[Rewrite Widget]] updated doc for distinctTitle property

livecodegeorgia committed:
[[Rewrite Widget]] updated fetchAndroidColors() handler to use the named colour constants for the keys in the array

livecodeali committed:
Merge pull request #2923 from livecodeali/bugfix-16009

[[ Bug 16009 ]] Menu arrays come from script, so default values must be strings

livecodefraser committed:
[[ RawClipboard ]] Windows: remove an un-needed class

livecodefraser committed:
[[ RawClipboard ]] Windows: fix dragging into LiveCode from other programs

livecodegeorgia committed:
Merge pull request #2933 from livecodegeorgia/feature-header_widget_rewrite

Feature header widget rewrite

Commit:
[[ Docs ]] Update lcb spec

livecodefraser committed:
[[ RawClipboard ]] Windows: support IStream data objects

livecodefraser committed:
[[ RawClipboard ]] Ensure that 'the keys of the clipboarddata' doesn't contain duplicates

livecodeali committed:
Merge pull request #2938 from livecodeali/lcb-spec_update

[[ Docs ]] Update lcb spec

livecodefraser committed:
Merge branch 'develop' of https://github.com/runrev/livecode into feature-raw_clipboard

Conflicts:
	engine/engine-sources.gypi
	engine/src/dispatch.cpp
	engine/src/dsklnx.cpp
	engine/src/executionerrors.h
	engine/src/lnxdce.cpp
	engine/src/lnxdclnx.cpp
	engine/src/lnxdnd.cpp
	engine/src/lnxtransfer.cpp
	engine/src/object.h
	engine/src/parsedef.h
	engine/src/platform.cpp
	engine/src/transfer.h
	libfoundation/include/foundation-auto.h

livecodefraser committed:
[[ RawClipboard ]] Add documentation for 'rawClipboardData' and 'lock/unlock clipboard'

livecodefraser commented:
Closing as this is now targeted for 8.0.
